### DISCUSSION TITLE
Connect ESP32 displays (M5Stack and M5Stick devices) directly to xdrip through Bluetooth

### DISCUSSION THREAD

**Discussion by winni67** [NightscoutFoundation/xDrip#3793] [Ideas]
Hello, I found a cool feature in xdrip4ios, where it is possible to connect ESP32 displays (M5Stack and M5Stick Devices) directly to the xdrip phone trough Bluetooth to display the Glucose readings. Look at https://xdrip4ios.readthedocs.io/en/latest/connect/devices/
No local network or permanent cloud connection is necessary.

**Comment by Missdazzle** [comment]
That would be kind of nice to have!

**Comment by Navid200** [comment]
xDrip has a screensaver that displays your glucose and graph on screen when the phone is plugged in and being charged:
https://navid200.github.io/xDrip/docs/Screensaver.html 

If we can use something that xDrip already offers, it will be better because there will be no need to modify xDrip.
Can you tell me what is better in the device shown in your link compared to what xDrip offers already?
Is it because it has larger fonts?
Is it because you can see it easier if you wake up in the middle of the night with no need to raise in the bed to be able to see the phone?

There are phone stands that hold the phone on a nightstand.  This is just a screenshot of many options you can find with a search:
![Screenshot 2024-12-02 183915](https://github.com/user-attachments/assets/40564dc3-e218-44bb-94be-1173a50af42d)

**Reply by winni67** [reply]
In my use case, the best way is to put an external device outside on my boat, so that the crew is able to see whats going on with me. I think its no good idea to place the loop smartphone outside of the boot and expose it to water, salt water, and the sun. At sea, there is often no mobile network, so a connection of the M5stack over the internet doesn’t work for me. My smartwatch is behind the jacket cuffs, so it is difficult to look at the smartwatch to see the glucose value. It is impossible for my crew to see the glucose value on my smartwatch too. A M5stack display, externally mounted on my boat is the best solution for all people on board including myself.
The best way in the future would be to integrate an ESP32 as NMEA2000 Gateway, to display the glucose value directly on the navigation displays. There is an ESP32 project, that is able to communicate directly to the NMEA2000 instruments, but first, it is necessary, to get the glucose data into the ESP32, perhaps via Bluetooth, as already realized in xdrip4ios.
https://github.com/AK-Homberger/NMEA2000-Workshop
Last year, I build the remote control based on another project from Homberger
https://github.com/AK-Homberger/ESP32-Evo-Remote-Pilot-NMEA2000

**Reply by anbraten** [reply]
This change should allow to send the current bg to a ble device / display: https://github.com/NightscoutFoundation/xDrip/compare/master...anbraten:ble-display?expand=1

I haven't finished the auth part with the original m5 stack code yet, but that should be easy to add.

**Reply by Navid200** [reply]
That sounds like a perfect description why this will be useful.
My only concern is if this is a standard or not?  What I mean is what happens if in 6 months, another manufacturer comes out with another clock and everyone then prefers to use that one?  Will we then need to add yet another section to xDrip to support that other device?  And what happens if yet another device comes out later?

xDrip already has a tasker feature.  And I must say I have no experience with it.  Can that be used?

I personally see no problem with someone adding a feature for this request.
Please click on "Create issue from discussion" in the right pane.

**Reply by anbraten** [reply]
While "hacking" on this I've noticed as well that adjusting the app would probably be more difficult / messy than adjusting such things in diy esp32 projects. Thereby existing projects like a nmea2000 bridge could probably be easily adopted to use a generic ble broadcast from xdrip.

**Comment by andiww** [comment]
Definitely a cool thing and with Android it would also be very interesting!
